pylint/testutils/_primer/primer_run_command.py

```diff
@@ -8,63 +8,65 @@
 import sys
 import warnings
 from io import StringIO
-from itertools import chain
 
 from pylint.lint import Run
 from pylint.message import Message
 from pylint.reporters import JSONReporter
+from pylint.reporters.json_reporter import OldJsonExport
 from pylint.testutils._primer.package_to_lint import PackageToLint
-from pylint.testutils._primer.primer_command import PackageMessages, PrimerCommand
+from pylint.testutils._primer.primer_command import PrimerCommand
 
 GITHUB_CRASH_TEMPLATE_LOCATION = "/home/runner/.cache"
 CRASH_TEMPLATE_INTRO = "There is a pre-filled template"
 
 
 class RunCommand(PrimerCommand):
     def run(self) -> None:
-        packages: PackageMessages = {}
-
+        packages: dict[str, list[OldJsonExport]] = {}
+        astroid_errors: list[Message] = []
+        other_fatal_msgs: list[Message] = []
         for package, data in self.packages.items():
-            packages[package] = self._lint_package(data)
-            print(f"Successfully primed {package}.")
+            messages, p_astroid_errors, p_other_fatal_msgs = self._lint_package(
+                package, data
+            )
+            astroid_errors += p_astroid_errors
+            other_fatal_msgs += p_other_fatal_msgs
+            packages[package] = messages
+        plural = "s" if len(other_fatal_msgs) > 1 else ""
+        assert not other_fatal_msgs, (
+            f"We encountered {len(other_fatal_msgs)} fatal error message{plural}"
+            " that can't be attributed to bleeding edge astroid alone (see log)."
+        )
+        path = (
+            self.primer_directory
+            / f"output_{'.'.join(str(i) for i in sys.version_info[:3])}_{self.config.type}.txt"
+        )
+        print(f"Writing result in {path}")
+        with open(path, "w", encoding="utf-8") as f:
+            json.dump(packages, f)
 
+    def _filter_astroid_errors(
+        self, messages: list[OldJsonExport]
+    ) -> tuple[list[Message], list[Message]]:
+        """This is to avoid introducing a dependency on bleeding-edge astroid."""
```

Comment:
```suggestion
        """Separate fatal errors caused by astroid that are not caused by a change in pylint."""
```

Reply:
Oh yeah, I actually proved this myself 😓 

```suggestion
        """Separate fatal errors caused by astroid so we can report them independently."""
```